dynawo/3rdParty/libzip/FindlibZIP.cmake

```diff
@@ -15,53 +15,79 @@
-        MATH(EXPR LIBZIP_MINOR_VERSION "${_LIBZIP_VERSION_STRING} / 100 % 1000")
-        MATH(EXPR LIBZIP_SUBMINOR_VERSION "${_LIBZIP_VERSION_STRING} % 100")
-        SET(libZIP_VERSION "${LIBZIP_MAJOR_VERSION}.${LIBZIP_MINOR_VERSION}.${LIBZIP_SUBMINOR_VERSION}")
-    ENDIF ()
-ENDIF ()
+set(libZIP_VERSION "")
+if(libZIP_INCLUDE_DIR AND EXISTS "${libZIP_INCLUDE_DIR}/libzip/ZipVersion.h")
+  # The version string appears in the following in the header:
+  # #define  LIBZIP_VERSION 100300
+  set(_LIBZIP_VERSION_REGEX "^#define[ \t]+LIBZIP_VERSION[ \t]+([0-9]+)")
+  file(STRINGS "${libZIP_INCLUDE_DIR}/libzip/ZipVersion.h" _LIBZIP_VERSION_STRING LIMIT_COUNT 1 REGEX "${_LIBZIP_VERSION_REGEX}")
+  string(REGEX REPLACE ${_LIBZIP_VERSION_REGEX} "\\1" _LIBZIP_VERSION_STRING "${_LIBZIP_VERSION_STRING}")
+  if(_LIBZIP_VERSION_STRING)
+    math(EXPR LIBZIP_MAJOR_VERSION "${_LIBZIP_VERSION_STRING} / 100000")
+    math(EXPR LIBZIP_MINOR_VERSION "${_LIBZIP_VERSION_STRING} / 100 % 1000")
+    math(EXPR LIBZIP_SUBMINOR_VERSION "${_LIBZIP_VERSION_STRING} % 100")
+    set(libZIP_VERSION "${LIBZIP_MAJOR_VERSION}.${LIBZIP_MINOR_VERSION}.${LIBZIP_SUBMINOR_VERSION}")
+  endif()
+endif()
 
-MESSAGE("-- libZIP version : ${libZIP_VERSION}")
-IF (libZIP_FIND_VERSION)
-    IF ("${libZIP_VERSION}" VERSION_LESS "${libZIP_FIND_VERSION}")
-        MESSAGE("\n\tDetected version of libZIP is too old. Requested version was ${libZIP_FIND_VERSION}.\n")
-        SET(libZIP_FOUND FALSE)
-    ENDIF ()
-ELSE ()
-    # Handle the QUIETLY and REQUIRED arguments and set libZIP_FOUND
-    # to TRUE if all listed variables are TRUE.
-    # (Use ${CMAKE_ROOT}/Modules instead of ${CMAKE_CURRENT_LIST_DIR} because CMake
-    #  itself includes this FindlibZIP when built with an older CMake that does
-    #  not provide it.  The older CMake also does not have CMAKE_CURRENT_LIST_DIR.)
-    INCLUDE(FindPackageHandleStandardArgs)
-    FIND_PACKAGE_HANDLE_STANDARD_ARGS(libZIP
-        REQUIRED_VARS libZIP_LIBRARY libZIP_INCLUDE_DIR
-        VERSION_VAR libZIP_VERSION)
-ENDIF ()
+message("-- libZIP version : ${libZIP_VERSION}")
+if(libZIP_FIND_VERSION)
+  if("${libZIP_VERSION}" VERSION_LESS "${libZIP_FIND_VERSION}")
+    message("\tDetected version of libZIP is too old. Requested version was ${libZIP_FIND_VERSION}.")
+    set(libZIP_FOUND FALSE)
+  else()
+    set(libZIP_FOUND TRUE)
+  endif()
+else()
+  # Handle the QUIETLY and REQUIRED arguments and set libZIP_FOUND
+  # to TRUE if all listed variables are TRUE.
+  # (Use ${CMAKE_ROOT}/Modules instead of ${CMAKE_CURRENT_LIST_DIR} because CMake
+  #  itself includes this FindlibZIP when built with an older CMake that does
+  #  not provide it.  The older CMake also does not have CMAKE_CURRENT_LIST_DIR.)
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(libZIP
+    REQUIRED_VARS libZIP_LIBRARY libZIP_INCLUDE_DIR
+    VERSION_VAR libZIP_VERSION)
+endif()
+
+if(libZIP_FOUND)
```

Comment:
This code wasn't here before if I'm not mistaken. When a coding style change creates so many differences, it is always better to not add new code (even if seems harmless). Please put this change in another PR (along with the one in libArchive)

Reply:
in fact those changes are just copy/paste of the same file in dynawo/cmake, we had the same files but not containing the same code so I just corrected this. so it is not really new code